src/coap_io_posix.c

```diff
@@ -247,7 +249,16 @@ coap_io_process_with_fds_lkd(coap_context_t *ctx, uint32_t timeout_ms,
   int result;
   unsigned int i;
 #endif /* ! COAP_EPOLL_SUPPORT */
-
+#if ! COAP_CONSTRAINED_STACK && ! COAP_EPOLL_SUPPORT
+  fd_set readfds, writefds, exceptfds;
+#define COAP_READFDS readfds
+#define COAP_WRITEFDS writefds
+#define COAP_EXCEPTFDS exceptfds
+#else
+#define COAP_READFDS (ctx->readfds)
+#define COAP_WRITEFDS (ctx->writefds)
+#define COAP_EXCEPTFDS (ctx->exceptfds)
+#endif /* ! COAP_CONSTRAINED_STACK */
```

Comment:
This should include && ! COAP_EPOLL_SUPPORT.

If this #if block was to be moved up 2 lines to be inside the previous `#endif /* ! COAP_EPOLL_SUPPORT */`, then things would be cleaner.